Add bleeding column backgrounds to ffe-grid(-react)

selbekk opened the pull request:
This commit adds support for adding background colors to columns. These background colors will bleed out into their surrounding gutters, to make the content stand out more.

Notice that this is implemented in LESS by simply copying a lot of the `ffe-grid__row--bg-` classes. I considered pulling these out into their own helper classes, but that would've been quite a breaking change for many consumers - without any real gain (well, a few bytes less CSS I guess).

Fixes #197

Commit:
feat(ffe-grid): Add bleeding column backgrounds

This commit adds the concept of backgrounds on ffe-grid columns. It's
implemented by adding the same 8 `--bg-[bg-color]` class modifiers that
are supported by grid rows.

wkillerud commented on packages/ffe-grid-react/src/background-colors.js:
```diff
@@ -0,0 +1,10 @@
+    'grey-cloud',
+    'purple-magenta',
+    'sand',
+];
```
Prettier doesn't add an empty newline at EOF? :police_car:

selbekk replied:
Doesn't look like it - but the linter should at least warn. It doesn't though. Anyhow, fixed!

wkillerud reviewed (dismissed):
I'd love some screenshots of the result

wkillerud reviewed (changes requested):
Looks like some old temporary files from ` packages/ffe-icons-react/tmp/*` were commited.

selbekk commented:
Ah god damn it. I'll make sure to add them to the gitignore file as well

Commit:
chore: Ignore tmp files from git

Commit:
feat(ffe-grid-react): Add support for col bg

Add support for columns to have bleeding backgrounds. A new prop
`background` is provided, which accepts the same background color name
values as `GridRow`.

ivarni commented on packages/ffe-grid-react/src/Grid.md:
```diff
@@ -58,5 +58,37 @@ const backgroundColors = [
+        <GridRow key={idx}>
+            <GridCol
+                sm={12}
+                md={{ cols: 6, offset: 3}}
```
Offsett 3 på `md` er vel litt fyfy? Syns vi skal holde oss til våre [egne standarder](https://github.com/SpareBank1/designsystem/blob/develop/packages/ffe-grid-react/src/utils.dev.js#L42-L45) i dokumentasjonen :)

selbekk replied:
Uff da - fixed!

selbekk commented:
Here's a screenshot. I've added a minor improvement to the documentation so that it's easier to look at how rows and columns will look with different background colors:

![image](https://user-images.githubusercontent.com/1307267/39213992-5976f6c6-4813-11e8-9a01-25a4907d00b0.png)

selbekk committed:
docs(ffe-grid-react): Improve background examples

selbekk committed:
Merge branch 'develop' into 197-bleeding-column-backgrounds
